### ISSUE TITLE
Generated cluster id for kraft mode is sometimes invalid

### ISSUE THREAD

**Issue by fedinskiy** [strimzi/test-container#72]
When running a container in kraft mode, I sometimes (approximately once in 100 runs) see the container fail with status 1 and the following error in the logs:

```
2024-07-04T15:24:09.5338779Z usage: kafka-storage format [-h] --config CONFIG --cluster-id CLUSTER_ID
2024-07-04T15:24:09.5339351Z                      [--add-scram ADD_SCRAM] [--ignore-formatted]
2024-07-04T15:24:09.5339892Z                      [--release-version RELEASE_VERSION]
2024-07-04T15:24:09.5340442Z kafka-storage: error: argument --cluster-id/-t: expected one argument
2024-07-04T15:24:09.5341500Z [2024-07-04 15:19:11,430] INFO Registered kafka:type=kafka.Log4jController MBean (kafka.utils.Log4jControllerRegistration$)
2024-07-04T15:24:09.5343120Z [2024-07-04 15:19:11,654] INFO Setting -D jdk.tls.rejectClientInitiatedRenegotiation=true to disable client-initiated TLS renegotiation (org.apache.zookeeper.common.X509Util)
2024-07-04T15:24:09.5344263Z [2024-07-04 15:19:11,722] ERROR Exiting Kafka due to fatal exception (kafka.Kafka$)
2024-07-04T15:24:09.5344897Z java.lang.RuntimeException: No readable meta.properties files found.
2024-07-04T15:24:09.5345714Z 	at org.apache.kafka.metadata.properties.MetaPropertiesEnsemble.verify(MetaPropertiesEnsemble.java:493)
2024-07-04T15:24:09.5346567Z 	at kafka.server.KafkaRaftServer$.initializeLogDirs(KafkaRaftServer.scala:152)
2024-07-04T15:24:09.5347197Z 	at kafka.server.KafkaRaftServer.<init>(KafkaRaftServer.scala:60)
2024-07-04T15:24:09.5347787Z 	at kafka.Kafka$.buildServer(Kafka.scala:82)
2024-07-04T15:24:09.5348162Z 	at kafka.Kafka$.main(Kafka.scala:90)
2024-07-04T15:24:09.5348486Z 	at kafka.Kafka.main(Kafka.scala)
2024-07-04T15:24:09.5348701Z 
```

My investigation showed, that this may happen when generated cluster id[1] looks like this: `-Mk3vxQVTc-iWEuxW2zREA`.  It looks like custom uuid generation code sometimes create invalid values.

I suggest the following:
1. Add `set -euv` to the generated bash script(`/testcontainers_start.sh`), so this code can fail early and users will more info dfor debugging
2. Fix the custom uuid generated code or replace its usage with a call to the existing kafka code ("bin/kafka-storage.sh random-uuid")

[1] https://github.com/strimzi/test-container/blob/main/src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java#L280

**Comment by scholzj** [comment]
@see-quick Could this be the same as here: https://github.com/strimzi/strimzi-kafka-operator/pull/9301?

**Comment by see-quick** [comment]
> @see-quick Could this be the same as here: [strimzi/strimzi-kafka-operator#9301](https://github.com/strimzi/strimzi-kafka-operator/pull/9301)?

Yeah, it seems like that. Let me fix it in the code.

**Comment by see-quick** [comment]
I have created a PR [1]

[1] -  https://github.com/strimzi/test-container/pull/73/files

**Comment by fedinskiy** [comment]
@see-quick is there a planned date for 0.107.0 release?

**Comment by rsvoboda** [comment]
@scholzj do you have any plans for 0.107 release? We are considering if we should wait for the release or disable the test for now.

**Comment by scholzj** [comment]
@see-quick Can we do a new release to fix this?

**Comment by see-quick** [comment]
> @scholzj do you have any plans for 0.107 release? We are considering if we should wait for the release or disable the test for now.

Hi, I am gonna do an RC today.

**Comment by gtroitsk** [comment]
@see-quick thank you, our tests works with pre-released [0.107.0-rc1](https://github.com/strimzi/test-container/releases/tag/0.107.0-rc1). When can we expect release?

**Comment by scholzj** [comment]
@gtroitsk He is off for the rest of this week. So the GA will likely happen next week.

**Comment by see-quick** [comment]
Hi @gtroitsk, yesterday I have released the [0.107.0](https://github.com/strimzi/test-container/releases/tag/0.107.0) version of strimzi test container, which contains a fix.
